From de9535397fd88723b73fa6e1ff16ad201280c7c4 Mon Sep 17 00:00:00 2001
From: Antonin Godard <antonin.godard@bootlin.com>
Date: Wed, 13 Nov 2024 13:48:41 +0100
Subject: [PATCH 3/3] u-boot: enable saving the env in fat partition

This is not enabled by default on the Beagleplay. Actually, CMD_SAVEENV
is defined but we only have ENV_IS_NOWHERE set so U-boot disables
saveenv. So this patch does the following:

- Set CONFIG_ENV_IS_IN_FAT and related variables.
- Unset CONFIG_ENV_IS_NOWHERE so that when U-boot fails to read the
  file uboot.env on initial boot (the file is not present yet), it
  doesn't fail to "save to nowhere" and save the env to uboot.env
  instead.

Signed-off-by: Antonin Godard <antonin.godard@bootlin.com>
---
 .../u-boot/u-boot-bb.org-2025.10+git/env-in-fat.cfg        | 7 +++++++
 meta-beagle/recipes-bsp/u-boot/u-boot-bb.org_2025.10.bb    | 2 ++
 2 files changed, 9 insertions(+)
 create mode 100644 meta-beagle/recipes-bsp/u-boot/u-boot-bb.org-2025.10+git/env-in-fat.cfg

diff --git a/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org-2025.10+git/env-in-fat.cfg b/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org-2025.10+git/env-in-fat.cfg
new file mode 100644
index 000000000000..a22b7db67f44
--- /dev/null
+++ b/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org-2025.10+git/env-in-fat.cfg
@@ -0,0 +1,7 @@
+CONFIG_ENV_IS_IN_FAT=y
+CONFIG_ENV_FAT_INTERFACE="mmc"
+CONFIG_ENV_FAT_DEVICE_AND_PART="1:1"
+CONFIG_ENV_FAT_FILE="uboot.env"
+CONFIG_SYS_MMC_ENV_DEV=0
+CONFIG_SYS_MMC_ENV_PART=0
+# CONFIG_ENV_IS_NOWHERE is not set
diff --git a/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org_2025.10.bb b/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org_2025.10.bb
index 1c4f069133c5..069c92bc55ce 100644
--- a/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org_2025.10.bb
+++ b/meta-beagle/recipes-bsp/u-boot/u-boot-bb.org_2025.10.bb
@@ -11,6 +11,8 @@ UBOOT_GIT_PROTOCOL = "https"
 BRANCH = "v2025.10-Beagle"
 SRCREV = "6825d60bea17774358f4199c4c7062801cfb931e"
 
+SRC_URI += "file://env-in-fat.cfg"
+
 BRANCH:pocketbeagle2 = "v2025.10-am62-pocketbeagle2"
 BRANCH:pocketbeagle2-k3r5 = "v2025.10-am62-pocketbeagle2"
 SRCREV:pocketbeagle2 = "4e0b6294a1185029d34eff6ed91f3ad1734b643f"
-- 
2.43.0

